File: src/test/java/sirius/kernel/commons/MultiMapTest.java
package sirius.kernel.commons;

import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class MultiMapTest {
    @Test
    public void test() {
        MultiMap<String, String> mm = MultiMap.create();
        mm.put("A", "A");
        mm.put("A", "B");
        assertArrayEquals(new String[]{"A", "B"}, mm.get("A").toArray(new String[mm.get("A").size()]));
        mm.put("A", "B");
        assertArrayEquals(new String[]{"A", "B", "B"}, mm.get("A").toArray(new String[mm.get("A").size()]));
        mm.remove("A", "B");
        assertArrayEquals(new String[]{"A"}, mm.get("A").toArray(new String[mm.get("A").size()]));
        mm.put("B", "A");
        mm.put("B", "C");
        assertArrayEquals(new String[]{"A", "B"}, mm.keySet().toArray(new String[mm.keySet().size()]));
        assertArrayEquals(new String[]{"A", "A", "C"}, mm.values().toArray(new String[mm.values().size()]));
        mm.getUnderlyingMap().clear();
        assertArrayEquals(new String[0], mm.get("A").toArray(new String[mm.get("A").size()]));
        mm.put("B", "C");
        assertArrayEquals(new String[]{"C"}, mm.get("B").toArray(new String[mm.get("B").size()]));
        mm.clear();
        assertArrayEquals(new String[0], mm.get("B").toArray(new String[mm.get("B").size()]));
    }
}

File: src/test/kotlin/sirius/kernel/commons/MultiMapTest.kt
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.kernel.commons

import org.junit.jupiter.api.Test
import kotlin.test.assertContentEquals

/**
 * Tests the [MultiMap] class.
 */
class MultiMapTest {

    @Test
    fun test() {
        val underTest = MultiMap.create<String, String>()
        underTest.put("A", "A")
        underTest.put("A", "B")
        assertContentEquals(listOf("A", "B"), underTest["A"])
        underTest.put("A", "B")
        assertContentEquals(listOf("A", "B", "B"), underTest["A"])
        underTest.remove("A", "B")
        assertContentEquals(listOf("A"), underTest["A"])
        underTest.put("B", "A")
        underTest.put("B", "C")
        assertContentEquals(listOf("A", "B"), underTest.keySet())
        assertContentEquals(listOf("A", "A", "C"), underTest.values())
        underTest.underlyingMap.clear()
        assertContentEquals(listOf(), underTest["A"])
        underTest.put("B", "C")
        assertContentEquals(listOf("C"), underTest["B"])
        underTest.clear()
        assertContentEquals(listOf(), underTest["B"])
    }
}
